examples/calibration/plot_calibration_curve.py

```diff
@@ -5,130 +5,262 @@
+# typical transposed-sigmoid curve. Calibration of the probabilities of
+# :class:`~sklearn.naive_bayes.GaussianNB` with isotonic regression can fix
+# this issue as can be seen from the nearly diagonal calibration curve. Sigmoid
+# regression also improves calibration slightly,
+# albeit not as strongly as the non-parametric isotonic regression. This can be
+# attributed to the fact that we have plenty of calibration data such that the
+# greater flexibility of the non-parametric model can be exploited.
+#
+# Below we show the Brier score , precision, recall and F1 score (see
+# :ref:`User Guide <precision_recall_f_measure_metrics>`). Notice that
+# although calibration improves the Brier score (a metric composed of
+# calibration loss and refinement loss), it does not significantly
+# alter the prediction accuracy measures (precision, recall and F1 score).
+# This is because calibration should not significantly move at the location of
+# the decision threshold (at x = 0.5 on the graph). Calibration should
+# however, make the predicted probabilities more accurate and thus more useful
+# for making allocation decisions under uncertainty.
 
-    # Logistic regression with no calibration as baseline
-    lr = LogisticRegression(C=1.)
+index = {}
+brier = []
+precision = []
+recall = []
+f1 = []
 
-    fig = plt.figure(fig_index, figsize=(10, 10))
-    ax1 = plt.subplot2grid((3, 1), (0, 0), rowspan=2)
-    ax2 = plt.subplot2grid((3, 1), (2, 0))
+for i, (clf, name) in enumerate(clf_list):
+    clf.fit(X_train, y_train)
+    y_pred = clf.predict(X_test)
 
-    ax1.plot([0, 1], [0, 1], "k:", label="Perfectly calibrated")
-    for clf, name in [(lr, 'Logistic'),
-                      (est, name),
-                      (isotonic, name + ' + Isotonic'),
-                      (sigmoid, name + ' + Sigmoid')]:
-        clf.fit(X_train, y_train)
-        y_pred = clf.predict(X_test)
-        if hasattr(clf, "predict_proba"):
-            prob_pos = clf.predict_proba(X_test)[:, 1]
-        else:  # use decision function
-            prob_pos = clf.decision_function(X_test)
-            prob_pos = \
-                (prob_pos - prob_pos.min()) / (prob_pos.max() - prob_pos.min())
+    # Create DataFrame index
+    index[i] = name
+    # Store column data
+    brier.append(brier_score_loss(y_test, y_pred))
+    precision.append(precision_score(y_test, y_pred))
+    recall.append(recall_score(y_test, y_pred))
+    f1.append(f1_score(y_test, y_pred))
 
-        clf_score = brier_score_loss(y_test, prob_pos, pos_label=y.max())
-        print("%s:" % name)
-        print("\tBrier: %1.3f" % (clf_score))
-        print("\tPrecision: %1.3f" % precision_score(y_test, y_pred))
-        print("\tRecall: %1.3f" % recall_score(y_test, y_pred))
-        print("\tF1: %1.3f\n" % f1_score(y_test, y_pred))
+score_df = pd.DataFrame(
+    data={'Brier score': brier, 'Precision': precision, 'Recall': recall,
+          'F1': f1},
+    index=index,
+)
+score_df
```

Comment:
We cannot expect more than 3 meaningful decimals and it would further improve readability:

```suggestion
score_df.round(3)
```